src/VisualStudio/Core/Def/Implementation/TodoComments/VisualStudioTodoCommentsService.cs

```diff
@@ -191,10 +198,13 @@ public ImmutableArray<TodoCommentData> GetTodoItems(Workspace workspace, Documen
         /// <summary>
         /// Callback from the OOP service back into us.
         /// </summary>
-        public Task OnDocumentRemovedAsync(DocumentId documentId, CancellationToken cancellationToken)
+        public async Task OnDocumentRemovedAsync(DocumentId documentId, CancellationToken cancellationToken)
         {
-            _documentToInfos.TryRemove(documentId, out _);
-            return Task.CompletedTask;
+            var workQueue = await _workQueueSource.Task.ConfigureAwait(false);
```

Comment:
💭 I worry about the impact of this `await`. However, if `_workQueueSource` is not already complete, don't we know that `_documentToInfos` doesn't contain `documentId`? If so, we could just return as a NOP if the task is not already complete.

```suggestion
            if (!_workQueueSource.Task.IsCompleted)
                return;

            var workQueue = await _workQueueSource.Task.ConfigureAwait(false);
```

Reply:
I've gone with a different approach that simply leverages the same codepath we're taking today.